### PR TITLE
RD2: Update non-Elevate fields on Closed Elevate RD

### LINKED ISSUE #6650: Misleading error message is displayed when updating Closed Elevate Recurring Donation
As a user I want to be able to edit non-Elevate fields on a Closed Elevate Recurring Donation from RD Edit Form.

Details and Steps to Reproduce

Test Steps:

1. NPSP Org connected to Elevate Int1 environment
2. Navigate to NPSP org and create a new contact
3. Navigate to recurring donations page and try to create a new recurring donation for an amount with value "50" and for the contact created in step#2
4. Wait for the RD details to display
5. From the top action buttons on the page, click close RD. Close the RD by providing a reason "Financial Difficulty"
6. Wait for the page to populate the status closed
7. click 'Edit' from the Quick Action and update any non-elevate related field

Expected:
The update goes through

Actual:
The error message: "Cannot read property of 'null'" is displayed

### PR DESCRIPTION
**Pull request by andrewyu-salesforce**
[W-9580620](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000QmHkYAK/view)
Fixes an issue when editing a Closed Elevate Recurring Donation, a misleading error will display block the update

# Critical Changes

# Changes

# Issues Closed
 Fixes #6650
# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers
Allow users to edit Closed Elevate Recurring Donations when the fields changes are not related to Elevate Service.

# New Metadata
- Custom Label: RD2_ElevateRDCannotChangeElevateFields

# Deleted Metadata

### CODE REVIEW

File: force-app/main/default/classes/RD2_ValidationService.cls [hunk 1]
```diff
@@ -257,6 +258,29 @@ public with sharing class RD2_ValidationService {
         return errorRecords;
     }
 
+    private void validateCloseElevateReucrringDonation(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd, ErrorRecord errorCollection) {
```
Comment (bdvorachek) [inline force-app/main/default/classes/RD2_ValidationService.cls:261]:
```suggestion
    private void validateClosedElevateRecurringDonation(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd, ErrorRecord errorCollection) {
```

File: force-app/main/default/classes/RD2_ValidationService.cls [hunk 2]
```diff
@@ -257,6 +258,29 @@ public with sharing class RD2_ValidationService {
...
+        RD2_RecurringDonation recurringDonation = new RD2_RecurringDonation(rd);
+        if (!recurringDonation.isElevateRecord() || !recurringDonation.isClosed()) {
+            return;
+        }
+
+        Boolean isElevateFieldChange = rd.npe03__Contact__c != oldRd.npe03__Contact__c
```
Comment (bdvorachek) [inline force-app/main/default/classes/RD2_ValidationService.cls:267]: Can we extract this conditional into a separate method? 
`private Boolean isElevateFieldChanged(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd)`

**Review by bdvorachek — APPROVED** [review]

**Review by dfuller114 — APPROVED** [review]
looks good!

**Review by dfuller114 — APPROVED** [review]
additional commit looks good!

**Review by npsp-reedestockton — APPROVED** [review]
Nicely handled!

**Pull request merged**